app/controllers/collections_controller.rb

```diff
@@ -94,14 +96,19 @@ def view_shared
     render :show
   end
 
-  def save_shared
+  def save_shared # rubocop:disable Metrics/AbcSize
     return redirect_to user_messages_path(current_user), alert: t('.errors.already_member') if @collection.users.include? current_user
 
-    @collection.users << current_user
-    if @collection.save
-      redirect_to @collection, notice: t('.success_notice')
+    ActiveRecord::Base.transaction(requires_new: true) do
+      @collection.users << current_user
+      message = Message.received_by(current_user).find_by(param_type: 'collection', param_id: @collection.id)
+      message.mark_as_deleted(current_user)
+      message.save!
+    rescue ActiveRecord::ActiveRecordError
```

Comment:
I tend to be more specific and rather list the errors I'd expect here. If something else is raised, we will notice in the monitoring.

Reply:
I don't really expect any errors here. I just want to make sure, that the database is in a consistent state. The user-feedback was just a bonus. As far as I see, any error is unexpected here and should end up in the monitoring. Should I just remove the `rescue` then?

Reply:
I'd say so.

Reply:
done